### PR TITLE
Fixed configuration indexing issue

### PR DESCRIPTION
**Pull request by shariq1501**

### CODE REVIEW

File: src/frontend/src/app/configuration-manager/configuration-manager.component.ts [hunk 1]
```diff
@@ -8,60 +8,36 @@ import { ConfigurationService, AlertService } from '../services/index';
...
+        data.categories.forEach(element => {
+          this.getCategory(element.key, element.description);
         });
-
       },
       error => { console.log("error", error) });
   }
 
-  private getCategory(category_name: string): void {
+  private getCategory(category_name: string, description:string): void {
```
Comment (praveen-garg) [inline src/frontend/src/app/configuration-manager/configuration-manager.component.ts]: category_description as first arg

File: src/frontend/src/app/configuration-manager/configuration-manager.component.ts [hunk 2]
```diff
@@ -8,60 +8,36 @@ import { ConfigurationService, AlertService } from '../services/index';
...
     var categoryValues = [];
     this.configService.getCategory(category_name).
       subscribe(
       data => {
         categoryValues.push(data);
-        this.categoryData.push({ key: category_name, value: categoryValues })
+        this.categoryData.push({description: description,key: category_name, value: categoryValues})
```
Comment (praveen-garg) [inline src/frontend/src/app/configuration-manager/configuration-manager.component.ts]: key, description, value or key, value, description

and space after comma

File: src/frontend/src/app/configuration-manager/configuration-manager.component.ts [hunk 3]
```diff
@@ -8,60 +8,36 @@ import { ConfigurationService, AlertService } from '../services/index';
...
-  //     data => {
-  //       console.log("data ", data)
-  //       if (data.value == "") {
-  //         // Hot fix to bind DOM element with latest updated values
-  //         location.reload();
-  //       }
-  //     },
-  //     error => { console.log("error", error) });
-  // }
-
```
Comment (praveen-garg) [inline src/frontend/src/app/configuration-manager/configuration-manager.component.ts:64]: thanks for removing this

**Review by praveen-garg — CHANGES_REQUESTED** [review]
save button color should be "primary"

**Comment by shariq1501** [comment]
@praveen-garg Please review now.

File: src/frontend/src/app/configuration-manager/configuration-manager.component.ts [hunk 4]
```diff
@@ -8,60 +8,36 @@ import { ConfigurationService, AlertService } from '../services/index';
...
     this.configService.getCategories().
       subscribe(
       data => {
-        this.configurationData = data.categories;
-        console.log("This is the congfigurationData ", this.configurationData);
-        this.configurationData.forEach(element => {
-          this.getCategory(element.key);
+        console.log("This is the congfigurationData ", data.categories);
+        data.categories.forEach(element => {
+          this.getCategory(element.key, element.description);
```
Comment (praveen-garg) [inline src/frontend/src/app/configuration-manager/configuration-manager.component.ts]: order x, y and accepted as y, x?

**Review by praveen-garg — CHANGES_REQUESTED** [review]

**Comment by praveen-garg** [comment]
@mshariq-nerd  please rebase with frontend

File: src/frontend/src/app/configuration-manager/configuration-manager.component.html [hunk 5]
```diff
@@ -20,41 +20,19 @@
               <div class="control">
                 <div class="media">
                   <div class="media-content">
-                    <p class="title is-5">{{v?.key}}</p>
```
Comment (praveen-garg) [inline src/frontend/src/app/configuration-manager/configuration-manager.component.html]: don't delete; just comment it

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**